### PR TITLE
FIX ElasticaService::define() now allows index to be deleted before (re)defining it again

### LINKED ISSUE #19: Cannot change mapping types without fully clearing the index
Hi team,

In some situation you might not have access to the ElasticSearch instance to delete the index and recreate it. Example: SilverStripe Platform.

One example is our addons.silverstripe.org website. Here are some steps to outline [the issue I'm having with it](https://github.com/silverstripe/addons.silverstripe.org/issues/241):

(`addons` is my index name)

* `curl -XDELETE http://localhost:9200/addons` to delete the index
* Run `dev/tasks/Heyday-Elastica-ReindexTask` to set initial index schema
* `$ curl -s http://localhost:9200/addons/Addon/_mapping | jq .addons.mappings.Addon.properties.downloads` (type: string)
* Edit `Addon::getElasticaMapping()` and change `downloads` from `string` to `long`
* Run `dev/tasks/Heyday-Elastica-ReindexTask` _to update index spec_

Error occurs here:

```
[Emergency] Uncaught Elastica\Exception\ResponseException: mapper [downloads] of different type, current_type [text], merged_type [long]
```

If you have ES access you can manually clear the index to avoid this happening, but ideally the reindex task in this module would do that for us.

### PR DESCRIPTION
**Pull request by robbieaverill**
This fixes a bug where changing mapping types on a DataObject after the index has been created would result in a fatal error, but allowing you to explicitly delete the index before redefining it again.

Fixes https://github.com/heyday/silverstripe-elastica/issues/19

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -36,9 +36,13 @@
         "ruflin/elastica": "~5.3",
         "symbiote/silverstripe-queuedjobs": "~4.0"
     },
+    "require-dev": {
```
Comment (sminnee) [inline composer.json:39]: Should we add a .travis.yml while we're at it? ;-)

**Review by sminnee — COMMENTED** [review]
1 optional suggestion

File: composer.json (hunk 1 above)
Comment (robbieaverill) [reply composer.json:39]: Done, example build via our fork here: https://travis-ci.org/creative-commoners/silverstripe-elastica/builds/515513180

**Comment by stevie-mayhew** [comment]
@tractorcow do you mind giving this a review when you have time?

**Comment by tractorcow** [comment]
Looks good to me.

**Comment by robbieaverill** [comment]
Thanks!

**Pull request merged**